packages/cli/src/util/dev/headers.ts

```diff
@@ -16,3 +16,60 @@ export function nodeHeadersToFetchHeaders(
   }
   return headers;
 }
+
+/**
+ * Adds/Updates/Deletes headers in `reqHeaders` based on the response headers
+ * from a middleware (`respHeaders`).
+ *
+ * `x-middleware-override-headers` is a comma-separated list of *all* header
+ * names that should appear in new request headers. Names not in this list
+ * will be deleted.
+ *
+ * `x-middleware-request-*` is the new value for each header. This can't be
+ * omitted, even if the header is not being modified.
+ *
+ * Headers listed in `skippedHeaders` will be ignored.
+ *
+ */
+export function applyOverriddenHeaders(
+  reqHeaders: { [k: string]: string | string[] | undefined },
+  respHeaders: Headers,
+  skippedHeaders: Set<string>
+) {
+  if (!respHeaders.has('x-middleware-override-headers')) {
+    return;
+  }
+
+  const overriddenHeaders: Set<string> = new Set();
+  for (const key of respHeaders
+    .get('x-middleware-override-headers')!
```

Comment:
Is there an edge case where `respHeaders.get('x-middleware-override-headers')` could return `null` or `undefined` or some that isn't a string?

Reply:
No there was a if guard to ensure it exists. Refactored that part so that we don't need to use `!` 👍